lib/Mojo/mysql/Database.pm

```diff
@@ -58,31 +58,13 @@ sub query {
   $self->$_ for qw(_next _watch);
 }
 
-sub _dequeue {
-  my ($self, $async, $query) = @_;
-
-  my $queue = $self->{queue} ||= [];
-  for (my $i = 0; $i <= $#$queue; $i++) {
-    my $sth = $queue->[$i];
-    return splice @$queue, $i, 1 if !(!$sth->{async} ^ !$async) && $sth->{Statement} eq $query;
-  }
-
-  return $self->dbh->prepare($query, $async ? {async => 1} : ());
-}
-
-sub _enqueue {
-  my ($self, $sth) = @_;
-  push @{$self->{queue}}, $sth;
-  shift @{$self->{queue}} while @{$self->{queue}} > $self->max_statements;
-}
-
 sub _next {
   my $self = shift;
 
   return unless my $next = $self->{waiting}[0];
   return if $next->{sth};
 
-  my $sth = $next->{sth} = $self->_dequeue(1, $next->{query});
+  my $sth = $next->{sth} = $self->dbh->prepare($next->{query}, {async => 1});
```

Comment:
Won't this make all statements async..? Doesn't look like it from the test results though. (There's something I'm missing here)